roofit/roofitcore/inc/RooDataHist.h

```diff
@@ -132,7 +132,11 @@ class RooDataHist : public RooAbsData, public RooDirItem {
   void SetName(const char *name) override;
   void SetNameTitle(const char *name, const char* title) override;
 
-  Int_t getIndex(const RooArgSet& coord, Bool_t fast = false) const;
+  Int_t getIndex(const RooAbsCollection& coord, Bool_t fast = false) const;
+  /// \copydoc getIndex(const RooAbsCollection&,Bool_t) const
+  Int_t getIndex(const RooArgSet& coord, Bool_t fast = false) const {
```

Comment:
What is the reason for keeping the `RooArgSet` version? Will the function with `RooAbsCollection` not accept a `RooArgSet` too because `RooAbsCollection` is the base class?

I guess you did this out of caution to not change the existing interface?

Reply:
I don't know. :-) That commit was a while ago, and nobody reviewed it since ...

I *think* it's because you can do an implicit conversion of
```c++
getIndex((RooRealVar)x);
```
to a RooArgSet, and this is unfortunately in use out there in the world. Now, since you cannot create a temporary `RooAbsCollection`, those instances will break, unless this overload is provided.
I'll check if that was the case, and then update the docs. Should I also hide it from doxygen to keep the function listings smaller?
@guitargeek

Reply:
And here you go:
```
../../../root-src/roofit/histfactory/src/ParamHistFunc.cxx:715:55: error: no viable conversion from 'const RooRealVar' to 'const RooAbsCollection'
    assert(static_cast<Int_t>(i) == _dataSet.getIndex(param)); // We assume that each parameter i belongs to bin i
```

Reply:
I see! Thanks for adding the explanation also in the source code. Otherwise I would have been worried that someone accidentally removes this function.